packages/api/src/controllers/playback.ts

```diff
@@ -44,11 +45,12 @@ const newPlaybackInfo = (
 const getAssetPlaybackUrl = async (
   config: Request["config"],
   ingest: string,
-  id: string
+  id: string,
+  user?: User
 ) => {
   const asset =
     (await db.asset.getByPlaybackId(id)) ??
-    (await db.asset.getByIpfsCid(id)) ??
+    (await db.asset.getByIpfsCid(id, user)) ??
```

Comment:
A flaw here is that if I pass my API key in the request I will get a 404 if the asset exists but is not owned by me. That asset that exists, even though is not in my account, is public, so IMO we should return it to anyone be them authenticated or not. In the future, we will support some kind of "public view" of an asset which is rate limited as well, which is what other users and anonymous callers would see. For now, we should just return the existing asset anyway.

So this logic needs to change to to use the asset from the user, if the user is authenticated and the asset exists, otherwise to return whatever playback ID already exists for that ID. This could be done either on the SQL query, maybe ordering based on the result being the ID we're looking for or not? Or maybe explicitly here, by doing separate queries first with the user then one without, in case of not found. We've already got a big list of queries here though and I'd not like to add 3 more now, so ideally we'd solve this on the SQL level.

WDYT?

Reply:
Mmm yeah in theory I would agree (even on long term) if the playbackUrl we provide when you pass the API key is a generic one and not one of another user. If we think this pr as a first step to let Studio track on demand usage, I would say the most important thing is to never provide playbackUrls of other users.

It could be a good idea now to do like this, and as a next step think about a system to generate a global separate playbackUrl for public assets in public view, that could be rate limited on the future or that can have a separate global usage.

Does it make sense for you?

Reply:
That would be a breaking change though, and I'm not sure what implications it would have. It also seems weird to me for an unauthenticated request to have more access than an authenticated one, incentivizing users to just use an unauthenticated player instead.

The authentication here is only in order to give preference to the users own assets when reading here, but it should not give them less content on their apps.

IMO we should not break this global playback URLs contract here, and continue allowing querying for any playback URL, as those should not be private. i.e. playback IDs should be all you need to build (or fetch) a playback URL
The work for the future would be giving support to these public rate limited playback URLs, but for now we should keep the current functionality working.

Reply:
Yeah, of course I agree we shouldn't introduce a breaking change on this.
For now, as you suggested, I added that when authenticated it returns the first available playbackUrl and added some tests for these new conditions